truffle/src/com.oracle.truffle.api.dsl.test/src/com/oracle/truffle/api/dsl/test/GenerateUncachedTest.java

```diff
@@ -209,7 +209,7 @@ public void testUncached5() {
         Uncached5Node.testAssumption.invalidate();
         assertEquals("s2", node.execute(42));
         Uncached5Node.testAssumption = null;
-        assertEquals("s1", node.execute(42));
+        assertEquals("s2", node.execute(42));
```

Comment:
This was a bug I fixed on the way. Uncached behavior was inconsistent with cached behavior.
In cached behavior it was not activating the specialization if the assumption was null, but in uncached behavior it did.
This was an bug caused by the fallback behavior where you need to check for null.
I don't expect this to have a big impact, as null assumption expressions are likely rare.